packages/zoe/src/zoeService/startInstance.js

```diff
@@ -1,51 +1,208 @@
+        const zoeSeatAdmin = state.seatHandleToSeatAdmin.get(seatHandle);
+        zoeSeatAdmin.replaceAllocation(allocation);
+      });
+    } catch (err) {
+      state.adminNode.terminateWithFailure(err);
+      throw err;
+    }
+  },
+  stopAcceptingOffers() {
+    const { state } = this;
+    return state.instanceAdmin.stopAcceptingOffers();
+  },
+  setOfferFilter(strings) {
+    const { state } = this;
+    state.instanceAdmin.setOfferFilter(strings);
+  },
+  getOfferFilter() {
+    const { state } = this;
+    return state.instanceAdmin.getOfferFilter();
+  },
+  getExitSubscriber(seatHandle) {
+    const { state } = this;
+    return state.seatHandleToSeatAdmin.get(seatHandle).getExitSubscriber();
+  },
+  isBlocked(string) {
+    const { state } = this;
+    return state.instanceAdmin.isBlocked(string);
+  },
+};
+
 /**
- * @param {MakeZoeInstanceStorageManager} makeZoeInstanceStorageManager
- * @param {UnwrapInstallation} unwrapInstallation
+ * @param {any} startInstanceAccess
  * @param {ERef<BundleCap>} zcfBundleCapP
  * @param {(id: string) => BundleCap} getBundleCapByIdNow
  * @param {Baggage} [zoeBaggage]
- * @returns {import('./utils.js').StartInstance}
+ * @returns {import('./utils').StartInstance}
  */
 export const makeStartInstance = (
-  makeZoeInstanceStorageManager,
-  unwrapInstallation,
+  startInstanceAccess,
   zcfBundleCapP,
   getBundleCapByIdNow,
   zoeBaggage = makeScalarBigMapStore('zoe baggage', { durable: true }),
 ) => {
   const makeInstanceHandle = defineDurableHandle(zoeBaggage, 'Instance');
-  const makeSeatHandle = defineDurableHandle(zoeBaggage, 'Seat');
+
+  /** @type {WeakStore<SeatHandle, ZoeSeatAdmin>} */
+  const seatHandleToZoeSeatAdmin = provideDurableWeakMapStore(
+    zoeBaggage,
+    'seatHandleToZoeSeatAdmin',
+  );
+
+  const instanceAdminMaker = makeInstanceAdminMaker(
+    zoeBaggage,
+    seatHandleToZoeSeatAdmin,
+  );
+
+  const makeZoeInstanceAdmin = vivifyFarClass(
+    zoeBaggage,
+    'zoeInstanceAdmin',
+    InstanceAdminI,
+    (instanceStorage, instanceAdmin, seatHandleToSeatAdmin, adminNode) => ({
+      instanceStorage,
+      instanceAdmin,
+      seatHandleToSeatAdmin,
+      adminNode,
+    }),
+    instanceAdminBehavior,
+  );
+
+  const vivifyEmptyFacet = facetName =>
+    vivifyFarClass(
+      zoeBaggage,
+      facetName,
+      M.interface(facetName, {}),
+      initEmpty,
+      {},
+    );
+  const makeEmptyCreatorFacet = vivifyEmptyFacet('emptyCreatorFacet');
+  const makeEmptyPublicFacet = vivifyEmptyFacet('emptyPublicFacet');
+
+  const makeAdminFacet = vivifyFarClass(
+    zoeBaggage,
+    'adminFacet',
+    AdminFacetI,
+    (adminNode, zcfBundleCap, contractBundleCap) => ({
+      adminNode,
+      zcfBundleCap,
+      contractBundleCap,
+    }),
+    {
+      getVatShutdownPromise() {
+        const { state } = this;
+
+        return E(state.adminNode).done();
+      },
+      restartContract(_newPrivateArgs) {
```

Comment:
I like to emphasize the optionality of parameters, for better local readability

```suggestion
      restartContract(_newPrivateArgs = undefined) {
```

Reply:
done